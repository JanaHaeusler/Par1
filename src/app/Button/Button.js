import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

const ButtonStyled = styled.button`
  padding: 7px;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  width: 80px;
  height: 34px;
  border-style: none;
  border-radius: 5px;
  box-shadow: 2px 2px 8px var(--secondary-medium-transparent);
  background-color: ${(props) =>
    props.main ? 'var(--primary)' : 'var(--light)'};
  color: ${(props) => (props.main ? 'var(--light)' : 'var(--primary)')};
  font-size: 1rem;

  &:disabled {
    opacity: 50%;
  }
`
const Button = ({ iconComponent, text, ...props }) => (
  <ButtonStyled {...props}>
    {iconComponent}
    {text}
  </ButtonStyled>
)

Button.propTypes = {
  main: PropTypes.bool,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  iconComponent: PropTypes.element.isRequired,
  text: PropTypes.string.isRequired,
}

export default Button
