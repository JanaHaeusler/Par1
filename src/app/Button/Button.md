Enabled Main Button

```jsx
import { CheckIconLight } from '../Icons/Icons'
;<Button
  main
  onClick={() => {}}
  iconComponent={<CheckIconLight />}
  text="Save"
/>
```

Disabled Main Button

```jsx
import { CheckIconLight } from '../Icons/Icons'
;<Button
  main
  onClick={() => {}}
  disabled
  iconComponent={<CheckIconLight />}
  text="Save"
/>
```

Enabled Button

```jsx
import { CheckIconDark } from '../Icons/Icons'
;<Button onClick={() => {}} iconComponent={<CheckIconDark />} text="Save" />
```
