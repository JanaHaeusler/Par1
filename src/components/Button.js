import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Button.propTypes = {
    children: PropTypes.string.isRequired,
  }

export default function Button({children}) {
    return (
        <ButtonStyled>{children}</ButtonStyled>
    )
}

const ButtonStyled = styled.button`
    padding: 7px;
    border-style: none;
    border-radius: 5px;
    box-shadow: 3px 3px 10px lightgrey;
    background-color: var(--primary);
    font-size: 1rem;
`