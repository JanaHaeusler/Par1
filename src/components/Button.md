Regular Button

```js
<Button>&#10003; Save</Button>
```

Disabled Button

```js
<Button disabled>&#10003; Save</Button>
```
