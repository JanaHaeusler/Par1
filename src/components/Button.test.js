import { render } from '@testing-library/react'

import Button from './Button'

describe('Button', () => {
    it('has the right text', () => {
        const { container } = render(<Button>Test Text</Button>)
        expect(container.firstChild).toMatchSnapshot()
    })
})