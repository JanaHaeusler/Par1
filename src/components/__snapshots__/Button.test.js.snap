// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has the right text 1`] = `
.c0 {
  padding: 7px;
  border-style: none;
  border-radius: 5px;
  background-color: #F5D87A;
  height: 50px;
  width: 100px;
  font-size: 1rem;
}

<button
  class="c0"
>
  âœ“ Save
</button>
`;
